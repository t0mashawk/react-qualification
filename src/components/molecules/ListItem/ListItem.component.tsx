import React from 'react'
import Link from '../../atoms/Link/Link.components'
import Button from '../../atoms/Button/Button.components'
import * as Styled from './ListItem.styles'

interface ListItemProps{
    text:string
    href:string
    buttonText: string
}

function ListItem(props: ListItemProps){
    const {text, href, buttonText} = props
    return(
        <Styled.Container>
            <Link href={href} text={text}/>
            <Button text={buttonText}/>
        </Styled.Container>
    )
}

export default ListItem