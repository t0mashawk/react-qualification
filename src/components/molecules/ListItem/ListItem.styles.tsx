import styled from "styled-components";

export const Container = styled.li`
    display:flex;
    background: lightblue;
    font-size:24px;
`