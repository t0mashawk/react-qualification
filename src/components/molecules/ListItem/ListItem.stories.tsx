import React from "react";
import { storiesOf } from "@storybook/react";
import ListItem from "./ListItem.component"

storiesOf("ListItem", module)
    .add("test",
        () => <ListItem text="link" href="http://www.google.com" buttonText="like"/>
    )