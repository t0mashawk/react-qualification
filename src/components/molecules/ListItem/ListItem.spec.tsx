import React from "react";
import renderer from "react-test-renderer"

import ListItem from "./ListItem.component"

test("LiListItem renders as expected", () => {
  const component = renderer.create(
    <ListItem text="Example Value" href="www.google.com" buttonText="example"/>
  );

  const tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});