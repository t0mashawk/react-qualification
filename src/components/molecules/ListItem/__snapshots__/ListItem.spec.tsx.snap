// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiListItem renders as expected 1`] = `
<li
  className="sc-AxirZ kGGelg"
>
  <a
    href="www.google.com"
  >
    Example Value
  </a>
  <div
    className="sc-AxjAm kNDLCK"
  >
    <button
      onClick={[Function]}
    >
      example
    </button>
  </div>
</li>
`;
